### DISCUSSION TITLE
decontX function in decontX package versus celda

### DISCUSSION THREAD

**Discussion by ccrobertson** [campbio/decontX#33] [General]
Hi there, 

Are there differences between the decontX function implemented in the standalone decontX R package compared to the function by the same name in the celda R package?

The decontX vignette on bioconductor (https://www.bioconductor.org/packages/release/bioc/vignettes/decontX/inst/doc/decontX.html) seems to include functions from celda that are not available in package decontX package. For example, plotDimReduceCluster. So if one chooses to load celda to have access to those functions, I'm wondering if there is a reason to load the decontX package and "decontX" function within that package instead of the one already available in celda. 

Many thanks for clarifying!

**Comment by joshua-d-campbell** [comment]
Hi @ccrobertson, thanks for your question and sorry for the confusion. There isn't any difference in the core functionality of the decontX functions between packages. We split up the code bases because the new decontPro algorithm for CITE-seq requires several new dependencies like rstan that we wanted to keep separate from the celda codebase. We are going to eventually migrate all of the decontX code to the decontX package and just have the Celda package import it. 

In short, you should be able to use the "decontX" function from either package without any issues. I'm going to move this to a Discussion so others can see it.

**Reply by ccrobertson** [reply]
Sorry for the delay. Thanks for the explanation!
